File: Lab03/C/primenumbers.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <ctype.h>

#include "primenumbers.h"

void compute_sieve(bool s[], unsigned n){
    unsigned i, j;
    for (i = 2; i <= n; i++){
        s[i] = true;
    }
    for (i = 2; i <= n; i++){
        if (s[i]){
            for(j = i + i; j <= n; j+=i){
                s[j] = false;
            }
        }
    }
}

unsigned count_primes(bool s[], unsigned n){
    int i, c = 0;
    for (i=2; i <= n; i++){
        if(s[i]) c++;
    }
    return c;
}



int main(int argc, char *argv[]){
    unsigned long n;
    unsigned long c;
    bool *s;
    char *endptr;


    if (argc != 2){
        printf("Zła liczba argumentów \n");
        return -1;
    }
    if (strtoul(argv[1], &endptr, 10) != 0){
        n = atol(argv[1]);
        printf("n: %lu\n", n);
        
        s = malloc((n+1)*sizeof(bool));
        compute_sieve(s,n);
        c = count_primes(s, n);
        free(s);
        printf("%lu\n", c);
        return 0;
    }
    else{
        printf("Błędne dane\n");
        return -1;
    }
    

}

File: Lab03/Python/packages/primenumbers.py
def create_sieve(s,n):
    for _ in range(n+1):
        s.append(True)
    s[0] = False
    s[1] = False
def compute_sieve(s):
    for i in range(2, len(s)):
        if s[i]:
            j = i + i
            while j < len(s):
                s[j] = False
                j = j + i
def count_primes(s):
    c = 0
    for e in s:
        if e:
            c = c + 1
    return c